src/ast/ast.h

```diff
@@ -456,31 +456,26 @@ char const * get_ast_kind_name(ast_kind k);
 
 class shared_occs_mark;
 
-class ast {
+class alignas(8) ast {  // Align to 8-byte boundary for efficient access
 protected:
     friend class ast_manager;
 
-    unsigned m_id = UINT_MAX;
-    unsigned m_kind:16;
-    // Warning: the marks should be used carefully, since they are shared.
+    // Cache-optimized layout: frequently accessed hot fields first
```

Comment:
all these fields are only 16 bytes, so they fit in a cache line. The order is irrelevant.